### PR TITLE
Handle winrm on windows

### PR DESCRIPTION
**Pull request by Annih**
Handle winrm transport on windows.
Refacto recipes to use the same base code for linux and windows.
Update chef spec accordingly

### CODE REVIEW

File: recipes/auth_key.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  file "#{node['rundeck_node']['home']}/.ssh/authorized_keys" do
+    owner     node['rundeck_node']['user']
+    group     node['rundeck_node']['group']
+    content   node['rundeck_node']['auth_public_key']
+    mode      '00600'
+    backup    false
+  end
+when 'windows'
+
+  include_recipe 'rundeck-node::local_user'
```
Comment (brugidou) [inline recipes/auth_key.rb]: Why do you need this? Couldn't we have an AD user with auth_key method?

File: recipes/local_user.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+if node['rundeck_node']['user_password_file']
+  if ::File.exist?(node['rundeck_node']['user_password_file'])
+    user_pwd = ::File.read(node['rundeck_node']['user_password_file'])
+  else
+    # include OpenSSL function secure_password
+    self.class.send(:include, Opscode::OpenSSL::Password)
+    user_pwd = secure_password
+  end
+
+  file node['rundeck_node']['user_password_file'] do
```
Comment (brugidou) [inline recipes/local_user.rb:32]: i think that you should enforce restrictive permissions to read this file

File: attributes/default.rb [hunk 3]
```diff
@@ -1,22 +1,45 @@
 #
-# Cookbook:   rundeck-node
-# Attributes: node
+# Author:: Robert Veznaver (<r.veznaver@criteo.com>)
+# Cookbook Name:: rundeck-node
+# Attribute:: default
 #
+# Copyright:: Copyright (c) 2015 Criteo.
```
Comment (rveznaver) [inline attributes/default.rb:6]: Please remove this. We already have a LICENSE file in the cookbook.

**Comment by rveznaver** [comment]
+1

**Pull request merged**